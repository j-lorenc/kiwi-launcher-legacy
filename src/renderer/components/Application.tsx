import { hot } from 'react-hot-loader/root';
import * as React from 'react';

export const Application: () => JSX.Element = () => {
  return <div>New Launcher</div>;
};

export default hot(Application);
