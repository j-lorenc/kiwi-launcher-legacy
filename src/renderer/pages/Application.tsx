import { hot } from 'react-hot-loader/root';
import React, { useReducer } from 'react';
import styles from './app.module.scss';
import HeaderBar from './HeaderBar/HeaderBar';
import FilterContext, { filterReducer, initialFilter } from '../contexts/filteredGame';
import SelectedGameContext, {
  selectedGameReducer,
  initialGameSelected,
  useSelectedGameContext,
} from '../contexts/selectedGame';
import WindowContext, {
  windowReducer,
  initialWindowSettings,
  useWindowContext,
} from '../contexts/currentWindow';

import FooterBar from './FooterBar/FooterBar';
import ContentSection from './ContentSection/ContentSection';

import { TransitionGroup, CSSTransition } from 'react-transition-group';
import { CurrentWindow } from '../../@types';

export const Application: () => JSX.Element = () => {
  const [state, dispatch] = useReducer(filterReducer, initialFilter);
  const [selectedGameState, selectedGameDispatch] = useReducer(
    selectedGameReducer,
    initialGameSelected
  );
  const [windowState, windowDispatch] = useReducer(windowReducer, initialWindowSettings);

  return (
    <FilterContext.Provider value={{ state, dispatch }}>
      <SelectedGameContext.Provider
        value={{ state: selectedGameState, dispatch: selectedGameDispatch }}
      >
        <WindowContext.Provider value={{ state: windowState, dispatch: windowDispatch }}>
          <div className={styles['app-container']}>
            <BackgroundImageContainer />
            <HeaderBar title="Kiwi Launcher" />
            <ContentSection />
            <FooterBar />
          </div>
        </WindowContext.Provider>
      </SelectedGameContext.Provider>
    </FilterContext.Provider>
  );
};

const BackgroundImageContainer: React.FC = () => {
  const { state: selectedGame } = useSelectedGameContext();
  const { state: windowState } = useWindowContext();

  if (windowState.currentWindow === CurrentWindow.HOME) return null;

  return (
    <TransitionGroup component={null}>
      <CSSTransition
        key={selectedGame ? `${selectedGame.id}-background` : ''}
        timeout={350}
        classNames={'fade'}
      >
        <div
          className={styles.background}
          style={selectedGame && { backgroundImage: `url(${selectedGame.backgroundUrl})` }}
        />
      </CSSTransition>
    </TransitionGroup>
  );
};

export default hot(Application);
