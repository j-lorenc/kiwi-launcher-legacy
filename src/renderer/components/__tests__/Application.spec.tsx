import * as React from 'react';
import { Application } from '../Application';
import * as renderer from 'react-test-renderer';

describe('Applcation', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<Application />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
