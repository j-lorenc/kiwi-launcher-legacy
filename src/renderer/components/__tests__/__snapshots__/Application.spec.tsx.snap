// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div>
  New Launcher
</div>
`;

exports[`Applcation should render correctly 1`] = `
<div>
  New Launcher
</div>
`;
